public_website/__tests__/fetchAPI.test.ts

```diff
@@ -0,0 +1,77 @@
+import { http, HttpHandler } from 'msw'
+import { describe, expect, it } from 'vitest'
+
+import { testDataFixtures } from './fixtures'
+import { server } from './server'
+import { fetchAPI } from '@/utils/fetchAPI'
+
+const respondWith = async (
+  body: unknown,
+  status = 200,
+  statusText?: string,
+  headers?: HeadersInit
+): Promise<Response> => {
+  return new Response(JSON.stringify(body), {
+    headers: {
+      'content-type': 'application/json',
+      ...headers,
+    },
+    status,
+    statusText,
+  })
+}
+
+describe('fetchAPI', async () => {
+  it('should fail if no token is found', async () => {
```

Comment:
C'est mieux avec "when" ? 

```suggestion
  it('should fail when no token is found', async () => {
```